interception/metadata/metadata_test.go

```diff
@@ -0,0 +1,61 @@
+package metadata
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+func TestUnaryClientInterceptor(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
+
+	key := "x-correlation-id"
+	value := "test-correlation-id"
+
+	ctx = metadata.NewOutgoingContext(ctx, metadata.Pairs(key, value))
```

Comment:
```suggestion
	ctx = metadata.NewIncomingContext(ctx, metadata.Pairs(key, value))
```

Should be always from Incoming to the outgoing